sky/clouds/service_catalog/gcp_catalog.py

```diff
@@ -551,11 +554,21 @@ def check_accelerator_attachable_to_host(instance_type: str,
 
 def get_image_id_from_tag(tag: str, region: Optional[str]) -> Optional[str]:
     """Returns the image id from the tag."""
-    return common.get_image_id_from_tag_impl(_image_df, tag, region)
+    global _image_df
+
+    image_id = common.get_image_id_from_tag_impl(_image_df, tag, region)
+    if image_id is None:
```

Comment:
Is it possible that the caller passed in an image tag that doesn't exist, and the current impl will keep re-pulling?

Reply:
This should only re-pull once, when the image tag is not found, which seems to be fine as for the failure case, it is ok to have some overheads. Wdyt?

Reply:
LGTM.